daemon/cmd/endpoint.go

```diff
@@ -177,13 +177,18 @@ func getEndpointIDHandler(d *Daemon, params GetEndpointIDParams) middleware.Resp
 // fetchK8sMetadataForEndpoint wraps the k8s package to fetch and provide
 // endpoint metadata. It implements endpoint.MetadataResolverCB.
 // The returned pod is deepcopied which means the its fields can be written
-// into.
-func (d *Daemon) fetchK8sMetadataForEndpoint(nsName, podName string) (*slim_corev1.Pod, *endpoint.K8sMetadata, error) {
+// into. Returns an error If a uid is given, and the uid of the retrieved
+// pod does not match it.
+func (d *Daemon) fetchK8sMetadataForEndpoint(nsName, podName, uid string) (*slim_corev1.Pod, *endpoint.K8sMetadata, error) {
 	p, err := d.endpointMetadataFetcher.FetchPod(nsName, podName)
 	if err != nil {
 		return nil, nil, err
 	}
 
+	if uid != "" && uid != string(p.GetUID()) {
+		return nil, nil, podStoreOutdatedErr
+	}
```

Comment:
A stricter approach would be moving this check to `fetchK8sMetadataForEndpointFromPod` below, so that it gets performed also after directly retrieving the pod object from k8s in the fallback path of `createEndpoint`.